merlin/models/torch/predict.py

```diff
@@ -0,0 +1,346 @@
+        ...
+
+    def __call__(self, data, batch_size=None, index=None, unique=True):
+        """Encode a Dataset, Loader, DataFrame, or Tensor(s).
+
+        Parameters
+        ----------
+        data : Dataset, Loader, DataFrameLike, torch.Tensor or Dict[str, torch.Tensor]
+            The data to be encoded.
+        batch_size : int, optional
+            The batch size for the encoding, by default None.
+        index : Optional[Selection], optional
+            The data selection used for the encoding, by default None.
+        unique : bool, optional
+            If True, duplicate rows in the DataFrame are removed, by default True.
+        """
+        if isinstance(data, (Dataset, Loader)):
+            return self.encode_dataset(data, batch_size, index=index, unique=unique)
+        if isinstance(data, DataFrameLike):
+            return self.encode_df(data, batch_size)
+        if isinstance(data, (dict, torch.Tensor)):
+            return self.encode_tensors(data)
+
+        raise ValueError("data must be a DataFrameLike, a Dataset, or a Loader")
+
+    def encode_dataset(
+        self,
+        data: Union[Dataset, Loader],
+        batch_size: Optional[int] = None,
+        index: Optional[Selection] = None,
+        unique: bool = True,
+    ) -> Dataset:
+        """Encode a Dataset or Loader through Dask.
+
+        Encoding happens in 3 steps:
+        1. Partition Mapping
+            This step uses Dask to break down the DataFrame into several partitions,
+            making large datasets computationally manageable.
+            The `call_df` function is applied to each partition independently,
+            facilitating efficient distributed computation.
+        2. DataFrame Processing
+            In this step, each partition, which is a DataFrame, is transformed directly
+            into a Loader with a determined batch size. This Loader then efficiently
+            converts the data into batches of PyTorch tensors, which are subsequently
+            processed by the PyTorch module using the `call_tensors` function.
+        3. Tensor Processing
+            Here, each batch derived from the Loader is processed by a PyTorch module for encoding.
+            If the inputs are dictionary-like and a passthrough_schema is provided, supplementary
+            columns might be included in the output DataFrame.
+
+        Parameters
+        ----------
+        data : Union[Dataset, Loader]
+            The data to be encoded.
+        batch_size : Optional[int], optional
+            The batch size for the encoding, by default None.
+        index : Optional[Selection], optional
+            The data selection used for the encoding, by default None.
+        unique : bool, optional
+            If True, duplicate rows in the DataFrame are removed, by default True.
+        """
+        if isinstance(data, Loader):
+            batch_size = data.batch_size
+            schema = data.input_schema
```

Comment:
Wondering if we need `data.output_schema` here, or does it not matter?

Reply:
You're right, output-schema tracking was not correct. Fixed it now